packages/components/src/circular-option-picker/option-picker-option.tsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useInstanceId } from '@wordpress/compose';
+import { useContext, useEffect } from '@wordpress/element';
+import { Icon, check } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { CircularOptionPickerContext } from './option-picker-context';
+import Button from '../button';
+import { CompositeItem } from '../composite';
+import Tooltip from '../tooltip';
+import type { OptionProps } from './types';
+
+const hasSelectedOption = new Map();
+
+function renderOptionAsButton( props: {
+	className?: string;
+	isPressed?: boolean;
+} ) {
+	return <Button { ...props }></Button>;
+}
+
+function renderOptionAsOption( props: {
+	id: string;
+	className?: string;
+	isSelected?: boolean;
+	compositeState: any;
+} ) {
+	const { className, isSelected, compositeState, ...additionalProps } = props;
+
+	return (
+		<CompositeItem
+			{ ...additionalProps }
+			{ ...compositeState }
+			as={ Button }
+			className={ classnames( className, {
+				'is-pressed': isSelected,
```

Comment:
Any reason why we're not passing `isPressed` as a prop here?

Reply:
Doing so [also sets `aria-pressed` on the control](https://github.com/WordPress/gutenberg/blob/b05bee7e2e8eefb2be06c228002e4f9783847747/packages/components/src/button/index.tsx#L143-L150), which we don't want for `role="option"`. I thought I'd tried to override that and failed, but clearly not hard enough. Definitely better to let `Button` worry about which classes to apply.

Reply:
Sounds good. Maybe let's add an inline comment above the `aria-pressed` override explaining the reason for it.

Reply:
Turns out I possibly had tried hard enough, because `aria-pressed={ null }` doesn't "work"...

```
Types of property ''aria-pressed'' are incompatible.
Type 'null' is not assignable to type 'boolean | "true" | "false" | "mixed" | undefined'.
```

Setting it to `null` does properly override the value, and prevent `aria-pressed` from showing up as an attribute in the HTML, but TypeScript doesn't like it.

Reply:
Unfortunately passing `undefined` as a value for `aria-pressed` gets it ignored entirely. I've added a note to explain the class addition.

> [!NOTE]
> **Follow-up task:** Update `Button` to check for `role` before setting `aria-pressed`, and set a more appropriate attribute (e.g. `aria-selected`) as needed.

Reply:
Sounds good — could you open a new issue about it? You could work on it as a follow-up in the upcoming weeks.